src/client/csfle/client_encryption.rs

```diff
@@ -639,3 +665,70 @@ impl From<String> for EncryptKey {
         Self::AltName(s)
     }
 }
+
+/// A pending `ClientEncryption::create_encrypted_collection` action.
+pub struct CreateEncryptedCollectionAction<'a> {
+    client_enc: &'a ClientEncryption,
+    db: Database,
+    name: String,
+    options: CreateCollectionOptions,
+    master_key: MasterKey,
+}
+
+impl<'a> CreateEncryptedCollectionAction<'a> {
+    /// Execute the collection creation.
+    pub async fn run(self) -> (Document, Result<()>) {
+        let ef = match self.options.encrypted_fields.as_ref() {
+            Some(ef) => ef,
+            None => {
+                return (
+                    doc! {},
+                    Err(Error::invalid_argument(
+                        "no encrypted_fields defined for collection",
+                    )),
+                );
+            }
+        };
+        let mut ef_prime = ef.clone();
+        if let Ok(fields) = ef_prime.get_array_mut("fields") {
+            for f in fields {
+                let f_doc = if let Some(d) = f.as_document_mut() {
+                    d
+                } else {
+                    continue;
+                };
+                if f_doc.get("keyId") == Some(&Bson::Null) {
+                    let d = match self
+                        .client_enc
+                        .create_data_key(self.master_key.clone())
+                        .run()
+                        .await
+                    {
+                        Ok(v) => v,
+                        Err(e) => return (ef_prime, Err(e)),
+                    };
+                    f_doc.insert("keyId", d);
+                }
+            }
+        }
+        let mut opts_prime = self.options.clone();
+        opts_prime.encrypted_fields = Some(ef_prime.clone());
+        (
+            ef_prime,
+            self.db.create_collection(&self.name, opts_prime).await,
+        )
+    }
+
+    /// Set the encrypted fields for the colllection.
+    pub fn encrypted_fields(mut self, encrypted_fields: impl Into<Option<Document>>) -> Self {
```

Comment:
Since it's an error (caught elsewhere) to call this method without `encrypted_fields`, it seemed worth giving this option its own action helper.

Reply:
Can this just accept a `Document` since it's an error for `None` to be passed?

Reply:
Good call, done.